### DISCUSSION TITLE
Lesson 4 : Error deploy.pi

### DISCUSSION THREAD

**Discussion by ForeManCrypto** [smartcontractkit/full-blockchain-solidity-course-py#1064] [Q&A]
Hi,

I have an error when install_solc("0.6.0")
I would appreciate any help.

Installing...
Traceback (most recent call last):
  File "C:\Users\XXX\web3\web3_py_simple_storage\deploy.py", line 8, in <module>
    install_solc("0.6.0")
  File "C:\Users\XXX\AppData\Roaming\Python\Python310\site-packages\solcx\install.py", line 456, in install_solc
    _install_solc_windows(version, filename, show_progress, solcx_binary_path)
  File "C:\Users\XXX\AppData\Roaming\Python\Python310\site-packages\solcx\install.py", line 627, in _install_solc_windows
    temp_path.rename(install_path)
  File "C:\Python310\lib\pathlib.py", line 1232, in rename
    self._accessor.rename(self, target)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\XXX\\AppData\\Local\\Temp\\solcx-tmp-16344' -> 'C:\\Users\\XXX\\.solcx\\solc-v0.6.0'

**Comment by n4n0b1t3** [comment]
Please check out the [chronological-issues-from-video.md for lesson 4](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#lesson-4)
If this already solves your problem please mark this post as answered.
If this does not solve the issue, we'd like to see more information about your code. Please refer to [how to improve your posts](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/980) before you write your question, this makes it easier for the community to help you.

**Comment by cromewar** [comment]
Hello @ForeManCrypto 
It seems you have a python installation as admin and is not allowing you to use the content of the python libraries due to a permission error, anyways as I said o post like [this](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1065#discussioncomment-2181001), pytnon 3.10 is a bad idea for web3 dev due to library conflicts.

Please reinstall python version 3.8 (not as admin) and run the code trough a python virtualenv.

Hint: Don't forget to select the correct python interpreter on vscode (bottom left corner).

**Reply by ForeManCrypto** [reply]
Hi, thanks for your help.
So I did what you explained. thanks a lot. I have 3.8
I changed path.
I still see this permission issue.

```
INFO: Could not find files for the given pattern(s).
Installing...
Traceback (most recent call last):
  File "deploy.py", line 8, in <module>
    install_solc("0.6.0")
  File "C:\Python\Python38\lib\site-packages\solcx\install.py", line 456, in install_solc
    _install_solc_windows(version, filename, show_progress, solcx_binary_path)
  File "C:\Python\Python38\lib\site-packages\solcx\install.py", line 627, in _install_solc_windows
    temp_path.rename(install_path)
  File "C:\Python\Python38\lib\pathlib.py", line 1340, in rename
    self._accessor.rename(self, target)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\XX\\AppData\\Local\\Temp\\solcx-tmp-20936' -> 'C:\\Users\\XX\\.solcx\\solc-v0.6.0'
```


The code is here:
```Python
from solcx import compile_standard, install_solc

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

# We add these two lines that we forgot from the video!
print("Installing...")
install_solc("0.6.0")


# Compile our Solidity


compiled_sol = compile_standard(
    {
        "language": "solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {
                    "*": ["abi", "metadata", "evm.bytecode", "evm.bytecode.sourceMap"]
                }
            }
        },
    },
    solc_version="0.6.0",
)
print(compiled_sol)
```

**Reply by ForeManCrypto** [reply]
I am using the .venv like you suggested but still I have this permission error when I tried deploy.py


INFO: Could not find files for the given pattern(s).
Installing...
Traceback (most recent call last):
  File "deploy.py", line 8, in <module>
    install_solc("0.6.0")
  File "C:\Users\XX\web3\web3_py_simple_storage\.venv\lib\site-packages\solcx\install.py", line 456, in install_solc
    _install_solc_windows(version, filename, show_progress, solcx_binary_path)
  File "C:\Users\XX\web3\web3_py_simple_storage\.venv\lib\site-packages\solcx\install.py", line 627, in _install_solc_windows
    temp_path.rename(install_path)
  File "C:\Python\Python38\lib\pathlib.py", line 1340, in rename
    self._accessor.rename(self, target)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\XX\\AppData\\Local\\Temp\\solcx-tmp-10608' -> 'C:\\Users\\XX\\.solcx\\solc-v0.6.0'
(.venv) PS C:\Users\XX\web3\web3_py_simple_storage>

**Reply by cromewar** [reply]
Three things
1. don't forget this line `"language": "solidity",` Solidity should be upper case
2. Are you installing python using Windows store? if so, please use the official installer from python page.
3. Does [this](https://stackoverflow.com/questions/26091530/permissionerror-winerror-5-access-is-denied-python-using-moviepy-to-write-gif) post help you?
